### PR TITLE
Fixed issues for the failing autodyne container and enabled web interface on host machine

### PR DESCRIPTION
**Pull request by Jonty16117**
This pr has the following changes:
- Fixed issues for the failing autodyne container
- Fixed "mkdir samples samples-output" issue
- Fixed python version issue
- Enabled web interface on host browser (added reverse proxy in containers)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ ENV TZ=America/New_York
 
 RUN sudo apt-get update && sudo apt-get -y upgrade
 
-RUN sudo apt-get install busybox-static fakeroot git tmux dmsetup kpartx netcat-openbsd nmap python-psycopg2 python3-psycopg2 libmagic1 python-lzma python-lzo liblzo2-dev python-six snmp uml-utilities util-linux vlan git unzip curl wget nano postgresql-client socat -y
+RUN sudo apt-get install busybox-static fakeroot git tmux dmsetup kpartx netcat-openbsd nmap python-psycopg2 python3-psycopg2 libmagic1 python-lzma python-lzo liblzo2-dev python-six snmp uml-utilities util-linux vlan git unzip curl wget nano postgresql-client socat nginx -y
```
Comment (compsecdirect) [inline Dockerfile:22]: Thanks for the PR.  
Why are you deploying a webserver though?  

You can use local ssh tunnels or any port redirector to allow for access to web-interfaces of the emulated devices.

**Review by compsecdirect — APPROVED** [review]

**Review by compsecdirect — DISMISSED** [review]

**Comment by Jonty16117** [comment]
webserver is just a nginx server serving the internal firmware router network on the container port 80

File: Dockerfile (hunk 1 above)
Comment (Jonty16117) [reply Dockerfile:22]: in my opinion that would be additional effort for anyone to setup the web-interface, in this implementation the user can just run docker-compose up -d and the web interface would be live on the go....

**Comment by Jonty16117** [comment]
as i said, its just simple nginx server serving on port 80, and that can be changed too from the nginx and docker-compose config

**Review by compsecdirect — CHANGES_REQUESTED** [review]
The nginx part should be completely separate from this project. Much of it is based on predictions, and hard-coding values into config files that should probably exist on another container.

File: README.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ Improve Firmadyne (https://github.com/firmadyne/firmadyne) and make it simpler t
 
 ## Usage
 
-1. Get firmware samples to analyze. ```mkdir samples samples-output``` Add firmware to ```samples``` folder
+1. Get firmware samples to analyze. Add firmware to ```samples``` folder
```
Comment (compsecdirect) [inline README.md:13]: Still need both ```samples ``` and ```samples-out```

File: docker-compose.yml [hunk 3]
```diff
@@ -81,9 +82,14 @@ services:
       - "/dev/loop11:/dev/loop11"
       - "/dev/loop12:/dev/loop12"
     command: [ "/opt/firmadyne/autodyne-0.5a.sh", "foo", "1.bin"]
+    environment:
+      - CONTAINER_NAME=autodyne-em1
+    ports:
```
Comment (compsecdirect) [inline docker-compose.yml:87]: Please remove ports

File: docker-compose.yml [hunk 4]
```diff
@@ -81,9 +82,14 @@ services:
...
+      - 3001:80
     volumes:
       - ./samples:/opt/firmadyne/samples
-      - ./sample_output:/opt/firmadyne/samples-out
+      - ./samples-out:/opt/firmadyne/samples-out
+      - ./nginx-config:/etc/nginx/sites-available
```
Comment (compsecdirect) [inline docker-compose.yml:92]: Please remove nginx

File: docker-compose.yml [hunk 5]
```diff
@@ -113,11 +119,16 @@ services:
       - "/dev/loop11:/dev/loop11"
       - "/dev/loop12:/dev/loop12"
     command: [ "/opt/firmadyne/autodyne-0.5a.sh", "netgear", "2.bin"]
+    environment:
+      - CONTAINER_NAME=autodyne-em2
+    ports:
```
Comment (compsecdirect) [inline docker-compose.yml:124]: Please remove ports

File: docker-compose.yml [hunk 6]
```diff
@@ -113,11 +119,16 @@ services:
...
+      - 3002:80
     volumes:
       - ./samples:/opt/firmadyne/samples
-      - ./sample_output:/opt/firmadyne/samples-out
+      - ./samples-out:/opt/firmadyne/samples-out
+      - ./nginx-config:/etc/nginx/sites-available
```
Comment (compsecdirect) [inline docker-compose.yml:129]: Please remove nginx

File: nginx-config/autodyne-em1/default [hunk 7]
```diff
@@ -0,0 +1,11 @@
+server {
```
Comment (compsecdirect) [inline nginx-config/autodyne-em1/default:1]: Please remove file

File: nginx-config/autodyne-em2/default [hunk 8]
```diff
@@ -0,0 +1,11 @@
+server {
```
Comment (compsecdirect) [inline nginx-config/autodyne-em2/default:1]: Please remove file

File: Dockerfile (hunk 1 above)
Comment (compsecdirect) [inline Dockerfile:22]: Please remove nginx

File: README.md [hunk 9]
```diff
@@ -20,8 +20,14 @@ b. manually increment the desired ip address.
 4. ```make build and make start```  
 5. ```docker exec CONTAINERID bash```  
 6. ```tmux ls```  
-7. ```tmux a -t "ImageID X```  Where X is the database id generated by firmadyne.  
+7. ```tmux a -t "ImageID X"```  Where X is the database id generated by firmadyne.  
 a. This tmux session is the console session to the firmware sample.
+8. To view the web interface on your host machine, follow these steps. Check the URL in the container logs. Look at line 6 in the file ```./nginx-config/{EMULATED_CONTAINER_NAME}/default```. If the URLs are different:
```
Comment (compsecdirect) [inline README.md:25]: Please remove nginx

File: autodyne-0.5a.sh [hunk 10]
```diff
@@ -19,6 +19,12 @@ cd /opt/firmadyne
 echo "Setting $Manufacturer for $FW"
 echo "Here is basename $BASENAME"
 
+# setup nginx
```
Comment (compsecdirect) [inline autodyne-0.5a.sh:22]: Please remove nginx

File: docker-compose.yml [hunk 11]
```diff
@@ -51,7 +51,8 @@ services:
     command: ["sh", "-c", "sleep 2073600"]
     volumes:
       - ./samples:/opt/firmadyne/samples
-      - ./sample_output:/opt/firmadyne/samples-out
+      - ./samples-out:/opt/firmadyne/samples-out
+      - ./nginx-config:/etc/nginx/sites-available
```
Comment (compsecdirect) [inline docker-compose.yml:55]: Please remove nginx

**Review by compsecdirect — CHANGES_REQUESTED** [review]

**Pull request closed without merging**